src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,262 @@
+    ancilla: RGN = RGN()
+
+    def __post_init__(self):
+        """Validation of the initialization arguments
+        
+        Constraints:
+        - the n values for the corners must be unique
+        - the n values for the corners must be in the interval ]0, ancilla.n[
+        - at least one betwween fields a and p in every corner is z
+        """
+        times = []
+        for rpng in self.corners:
+            if rpng.n:
+                if rpng.n < 1 or rpng.n >= self.ancilla.n:
+                    raise ValueError('The n values must be in ]0, ancilla meas time[.')
+                times.append(rpng.n)
+            # Confirm that at least one betwween fields a and p in every corner is z, if 2Q gate is present
+            if rpng.p and rpng.a != BasisEnum.Z and rpng.p != BasisEnum.Z:
+                raise ValueError('At least one between a and p must be "z" in every corner.')
+        if len(times) != len(set(times)):
+            raise ValueError('The n values for the corners must be unique.')
+        elif len(times) not in [0,2,4]:
+            raise ValueError('Each plaquette must have 0, 2, or 4 2Q gates.')
+
+
+    @classmethod
+    def from_string(cls, corners_rpng_string: str) -> 'RPNGDescription':
+        """Initialize the RPNGDescription object from a (16+3)-character string"""
+        rpng_objs = tuple([RPNG.from_string(s) for s in corners_rpng_string.split(' ')])
+        if len(rpng_objs) != 4:
+            raise ValueError('There must be 4 corners in the RPNG description.')
+        return cls(rpng_objs)
+    
+
+    @classmethod
+    def from_extended_string(cls, ancilla_and_corners_rpng_string: str) -> 'RPNGDescription':
+        """Initialize the RPNGDescription object from a (16+3)-character string"""
+        values = ancilla_and_corners_rpng_string.split(' ')
+        ancilla_rgn = RGN.from_string(values[0])
+        rpng_objs = tuple([RPNG.from_string(s) for s in values[1:]])
+        if len(rpng_objs) != 4:
+            raise ValueError('There must be 4 corners in the RPNG description.')
+        return cls(rpng_objs, ancilla_rgn)
+
+
+    def get_r_op(self, data_idx: int) -> str | None:
+        """Get the reset operation or Hadamard for the specific data qubit"""
+        return self.corners[data_idx].get_r_op()
+
+    def get_n(self, data_idx: int):
+        """Get the time of the 2Q gate involving the specific data qubit"""
+        return self.corners[data_idx].n
+
+    def get_g_op(self, data_idx: int) -> str | None:
+        """Get the measurement operation or Hadamard for the specific data qubit"""
+        return self.corners[data_idx].get_g_op()
+    
+    def get_plaquette(self, qubits: PlaquetteQubits = SquarePlaquetteQubits()) -> Plaquette:
+        """Get the plaquette corresponding to the RPNG description
+        
+        Note that the ancilla qubit is the last among the PlaquetteQubits and thus
+        has index 4, while the data qubits have indices 0-3.
```

Comment:
I think that currently the standard was to have the ancilla as the first qubit (index `0`). Is there any reason to change that?

Reply:
Also note that plaquettes on the boundary only have 3 qubits (counting the ancilla). Having a hole in qubit indexing should not be an issue, but I would advocate for not having a hole at all.

Reply:
That's a good point.
I just noticed that if I iterate through the `SquarePlaquetteQubits` the iterator first goes through the data qubits and then the ancilla qubit.
I do not know the consequences of changing that behavior, so I simply adopted that convention.
In the future one may have multiple ancillas to measure a stabilizer (e.g. when connectivity is constrained) so we may have other type of issees then).

Reply:
From memory, the iteration order is very much just a detail and is not used anywhere. A good hint that the previous sentence is likely true is that the iteration order is not documented anywhere. 

PS: jsut checked, `all_qubits` is only used to reduce a circuit to the plaquette's qubits by calling `ScheduledCircuit.filter_by_qubits`, so the specific ordering does not matter.

Reply:
Shall I change the order then (first ancilla, then data)?

Reply:
Ok, I just updated the `PlaquetteQubits` class changing the order of the fields (first `syndrome_qubits`, then `data_qubits`) and the `__iter__` method.

Reply:
@nelimee  This actually broke several CI/CD tests.
Shall we update those tests or change the way we iterate though the qubits in the `RPNGDescription.get_plaquette()` method?

I think it would be nice to have consistency across the codebase: either always ancilla first or data first

Reply:
> Shall we update those tests or change the way we iterate though the qubits in the `RPNGDescription.get_plaquette()` method?

For the moment, I would take the solution that minimise the amount of code changes for this PR. If that solution is not the prefered one (ancilla qubit first or last, I have a soft preference for ancilla qubit first) then we can open a different PR to change that.

So for the moment, stick to the convention that does not break tests and minimise code changes.

Reply:
Ok, I reverted the commit.
First data and then ancilla qubits.